src/MCUC.cpp

```diff
@@ -0,0 +1,136 @@
+    mcOn = gpios.mcStatusGPIO.readPin() == IO::GPIO::State::HIGH;
+
+    //set the inputs and step the model
+    VCU::MCuC_Model::ExtU_MCuC_Model_T inputs = {
+        ignitionOn,
+        startPressed,
+        brakeOn,
+        eStop,
+        forwardEnable,
+        mcState,
+        mcDischarge,
+        false,
+        hmFault,
+        throttle,
+        lvssOn,
+        mcOn,
+        powertrainCANSelfTestIn,
+        accessoryCANSelfTestIn};
+    model.setExternalInputs(&inputs);
+
+    model.step();
+
+    //get outputs
+    VCU::MCuC_Model::ExtY_MCuC_Model_T outputs = model.getExternalOutputs();
+    //save outputs
+    lvssEnable = outputs.LVSS_EN_uC;
+    inverterEnable = outputs.Inverter_EN_CAN;
+    ucFault = outputs.Fault;
+    watchdog = outputs.Watchdog;
+    ucState[0] = outputs.uC_State & 0b00000001;
+    ucState[1] = outputs.uC_State >> 1 & 0b00000001;
+    ucState[2] = outputs.uC_State >> 2 & 0b00000001;
+    ucState[3] = outputs.uC_State >> 3 & 0b00000001;
+    inverterDischarge = outputs.Inverter_DIS_CAN;
+    mcTogglePositive = outputs.MC_TOGp_uC;
+    mcToggleNegative = outputs.MC_TOGn_uC;
+    torqueRequest = outputs.Torque_Request;
+    mcSelfTestOut = outputs.MC_Self_Test;
+    estopSelfTestOut = outputs.ESTOP_Self_Test;
+    ignitionSelfTestOut = outputs.Ignition_Self_Test1;
+    accessoryCanSelfTestOut = outputs.CAN_A_out;
+    powertrainCanSelfTestOut = outputs.MC_CAN_out;
+
+    //use outputs
+    gpios.lvssEnableGPIO.writePin(lvssEnable ? IO::GPIO::State::HIGH : IO::GPIO::State::LOW);
+    //set inverterEnable before we send the message
+    gpios.ucFaultGPIO.writePin(ucFault ? IO::GPIO::State::HIGH : IO::GPIO::State::LOW);
+    gpios.watchdogGPIO.writePin(watchdog ? IO::GPIO::State::HIGH : IO::GPIO::State::LOW);
+    gpios.ucStateZeroGPIO.writePin(ucState[0] ? IO::GPIO::State::HIGH : IO::GPIO::State::LOW);
+    gpios.ucStateOneGPIO.writePin(ucState[1] ? IO::GPIO::State::HIGH : IO::GPIO::State::LOW);
+    gpios.ucStateTwoGPIO.writePin(ucState[2] ? IO::GPIO::State::HIGH : IO::GPIO::State::LOW);
+    gpios.ucStateThreeGPIO.writePin(ucState[3] ? IO::GPIO::State::HIGH : IO::GPIO::State::LOW);
+    //set inverterDisable before we send the message
+    gpios.mcTogglePositiveGPIO.writePin(mcTogglePositive ? IO::GPIO::State::HIGH : IO::GPIO::State::LOW);
+    gpios.mcToggleNegativeGPIO.writePin(mcToggleNegative ? IO::GPIO::State::HIGH : IO::GPIO::State::LOW);
+    //set torqueRequest before we send the message
+    gpios.mcSelfTestGPIO.writePin(mcSelfTestOut ? IO::GPIO::State::HIGH : IO::GPIO::State::LOW);
+    gpios.estopSelfTestGPIO.writePin(estopSelfTestOut ? IO::GPIO::State::HIGH : IO::GPIO::State::LOW);
+    gpios.ignitionSelfTestGPIO.writePin(ignitionSelfTestOut ? IO::GPIO::State::HIGH : IO::GPIO::State::LOW);
+    //We will send accessory CAN SelfTest message over CANopen
+    //Send the powertrainCanSelfTest message
+    if (powertrainCanSelfTestOut) {
+        powertrainCAN.sendUCSelfTestMessage();
```

Comment:
I was thinking that self test should be handled entirely before we get to the `process` loop